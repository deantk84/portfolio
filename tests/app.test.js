const request = require('supertest')
const app = require('../app')
 

test("app correctly errors on incorrect path", async () => {
    const response = await request(app).get('/incorrectpath')
    expect(response).toMatchSnapshot()
}) 

test("app correctly serves a home page", async () => {
    const response = await request(app).get('/')
    expect(response).toMatchSnapshot()
}) 