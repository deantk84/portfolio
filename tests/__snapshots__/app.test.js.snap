// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app correctly errors on incorrect path 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "152",
    "content-security-policy": "default-src 'none'",
    "content-type": "text/html; charset=utf-8",
    "date": "Fri, 08 Nov 2019 15:54:07 GMT",
    "x-content-type-options": "nosniff",
    "x-powered-by": "Express",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "user-agent": "node-superagent/3.8.3",
    },
    "method": "GET",
    "url": "http://127.0.0.1:62169/incorrectpath",
  },
  "status": 404,
  "text": "<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /incorrectpath</pre>
</body>
</html>
",
}
`;

exports[`app correctly serves a home page 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "12",
    "content-type": "text/html; charset=utf-8",
    "date": "Fri, 08 Nov 2019 15:54:07 GMT",
    "etag": "W/\\"c-Lve95gjOVATpfV8EL5X4nxwjKHE\\"",
    "x-powered-by": "Express",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "user-agent": "node-superagent/3.8.3",
    },
    "method": "GET",
    "url": "http://127.0.0.1:62171/",
  },
  "status": 200,
  "text": "Hello World!",
}
`;
